src/lib/telldus-accessory.js

```diff
@@ -164,15 +167,48 @@ class TelldusDimmer extends TelldusSwitch {
    */
   getDimLevel(callback, context) {
     this.log('getDimLevel called')
+    
     TDtool.device(this.id).then(device => {
       if (device.dimlevel)
-        return callback(null, bitsToPercentage(parseInt(device.dimlevel)))
+        this.dimlevel = bitsToPercentage(parseInt(device.dimlevel))
+        return callback(null, this.dimlevel)
       callback(null, device.lastsentcommand === 'ON' ? 100 : 0)
     })
 
   }
 
   /**
+   * Set the state of this Accessory. This accepts a value that is either
+   * true or false, for turning it on or off.
+   *
+   * @param  {*}                  value          The value to set,
+   *                                             corresponding to the passed
+   *                                             Characteristic
+   * @param  {Function}           callback       To be invoked when result is
+   *                                             obtained.
+   * @param  {object}             context
+   */
+  setState(value, callback, context) {
+    this.log(`Recieved set state request for dimmer: [${value ? 'on' : 'off'}]`)
+    if (value && this.dimlevel > 0){
+      TDtool.dim(percentageToBits(this.dimlevel), this.id).then(out => {
+        return out.indexOf('Success') > -1 ? callback() : Promise.reject(out)
+
+        // FIXME: This does not appear to actually be raising an error to
+        //        Homebridge, check out http://goo.gl/RGuILo . Same as above.
+      }, error => callback(new Error(error)))
+
+    }else{
+      TDtool['off'](this.id).then(out => {
```

Comment:
I believe that we can use 
```js
// Turn the dimmer of by using the super switch function
super.setState(false, callback, context)
``` 
to avoid duplication, and make this a bit more readable. This, in combination with using return in the if-case above, will allow us to avoid this else-case as a whole.